jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/FilesTask.java

```diff
@@ -0,0 +1,140 @@
+    JavaPluginConvention javaConvention =
+        project.getConvention().getPlugin(JavaPluginConvention.class);
+    SourceSet mainSourceSet = javaConvention.getSourceSets().findByName("main");
+    if (mainSourceSet == null) {
+      return;
+    }
+
+    // Print build.gradle
+    System.out.println(project.getBuildFile());
+
+    // Print settings.gradle
+    Path settingsFile = project.getProjectDir().toPath().resolve(Settings.DEFAULT_SETTINGS_FILE);
+    if (Files.exists(settingsFile)) {
+      System.out.println(settingsFile);
+    }
+
+    // Print sources
+    mainSourceSet.getAllJava().getSourceDirectories().forEach(System.out::println);
+
+    // Print resources
+    mainSourceSet.getResources().getSourceDirectories().forEach(System.out::println);
+
+    // Find project dependencies
+    for (Configuration configuration :
+        project.getConfigurations().getByName("runtime").getHierarchy()) {
+      for (Dependency dependency : configuration.getDependencies()) {
+        if (dependency instanceof ProjectDependency) {
+          // Keep track of project dependency jars
+          ProjectDependency projectDependency = (ProjectDependency) dependency;
+          String configurationName = projectDependency.getTargetConfiguration();
+          if (configurationName == null) {
+            configurationName = "default";
+          }
+          Project dependencyProject = projectDependency.getDependencyProject();
+          for (Configuration targetConfiguration :
+              dependencyProject.getConfigurations().getByName(configurationName).getHierarchy()) {
+            for (PublishArtifact artifact : targetConfiguration.getArtifacts()) {
+              projectDependencyJars.add(artifact.getFile());
+            }
+          }
+
+          // Print project dependency's build/source/resource files
+          listFilesForProject(dependencyProject, projectDependencyJars);
+        }
+      }
+    }
+  }
+
+  @Nullable private JibExtension jibExtension;
+
+  public FilesTask setJibExtension(JibExtension jibExtension) {
+    this.jibExtension = jibExtension;
+    return this;
+  }
+
+  @TaskAction
+  public void listFiles() {
+    Preconditions.checkNotNull(jibExtension);
+    Project project = getProject();
+
+    // If this is not the root project, print the root project's build.gradle and settings.gradle
+    if (project != project.getRootProject()) {
+      System.out.println(project.getRootProject().getBuildFile());
+      Path settingsFiles = project.getRootDir().toPath().resolve(Settings.DEFAULT_SETTINGS_FILE);
+      if (Files.exists(settingsFiles)) {
+        System.out.println(settingsFiles);
+      }
+    }
+
+    // Print extra layer
+    if (project.getPlugins().hasPlugin(JibPlugin.class)) {
+      System.out.println(jibExtension.getExtraDirectoryPath());
+    }
+
+    // Print sub-project sources
+    Set<File> skippedJars = new HashSet<>();
+    listFilesForProject(project, skippedJars);
```

Comment:
This will print the project's build and settings file again?

Reply:
No, the root build/settings files are only printed initially if the task is called on a sub-project.